src/com/mendix/widget/carousel/components/__tests__/Carousel.spec.ts

```diff
@@ -1,33 +1,95 @@
+import { CarouselItem } from "../CarouselItem";
 
 describe("Carousel", () => {
     const images: Image[] = [
         { url: "data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" },
         { url: "https://www.google.com/images/nav_logo242.png" }
     ];
-    let wrapper: ShallowWrapper<CarouselProps, any>;
-    beforeEach(() => wrapper = shallow(createElement(Carousel, { images })));
+    let output: ShallowWrapper<CarouselProps, any>;
+    let carouselWrapper: ShallowWrapper<CarouselProps, any>;
 
-    it("should be a 'div' of class carousel with a 'div' child of class carousel-inner", () => {
-        expect(wrapper).toMatchStructure(DOM.div({ className: "carousel" },
-            DOM.div({ className: "carousel-inner" }))
-        );
+    beforeEach(() => {
+        output = shallow(createElement(Carousel, { images }));
+        carouselWrapper = output.find("div.carousel-inner");
     });
 
-    it("should have the number of children equal to the number of images", () => {
-        expect(wrapper.find(".carousel-inner").children().length).toEqual(images.length);
+    it("renders a div of class carousel with a div child of class carousel-inner", () => {
+        const structure = DOM.div({ className: "carousel" },
+            DOM.div({ className: "carousel-inner" }));
+
+        expect(output).toMatchStructure(structure);
+    });
+
+    it("renders a wrapper with the same number of children as the images", () => {
+        let children = carouselWrapper.children();
+
+        expect(children.length).toEqual(images.length);
+    });
+
+    it("renders a wrapper with one child", () => {
+        output = shallow(createElement(Carousel, {
+                images: [ { url: "https://www.google.com/images/nav_logo242.png" } ]
+            }
+        ));
+
+        const children = output.find("div.carousel-inner").children();
+
+        expect(children.length).toEqual(1);
     });
 
-    it("should have the first image as active", () => {
-        expect(wrapper.find(".carousel-inner").children().first().prop("active")).toBe(true);
+    it("renders a wrapper with no children", () => {
+        output = shallow(createElement(Carousel));
+
+        const children = output.find("div.carousel-inner").children();
+
+        expect(children.length).toBe(0);
     });
 
-    it("should have only one active image", () => {
-        const children = wrapper.find(".carousel-inner").children();
+    it("renders a wrapper with the first child active", () => {
+        const firstChild = carouselWrapper.children().first();
+
+        expect(firstChild.prop("active")).toBe(true);
+    });
+
+    it("renders a wrapper with only one active child", () => {
+        const activeChildren = carouselWrapper.children().filterWhere(c => c.prop("active"));
+
+        expect(activeChildren.length).toBe(1);
+    });
+
+    describe("full component", () => {
+        let fullOutput: ReactWrapper<CarouselProps, any>;
```

Comment:
I'm doing a full mount for the next set of tests, so had to declare new variables for that.